### PR TITLE
Fix permissions on  secrets directory

### PR DESCRIPTION
**Pull request by rhatdan**
The permission on this directory will prevent user namespaces from using
secrets.  Should be world searchable

[NO NEW TESTS REQUIRED]

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
@ashley-cui PTAL
@nalind @umohnani8 @flouthoc @vrothberg @TomSweeneyRedHat PTAL

**Comment by ashley-cui** [comment]
LGTM

File: run_linux.go [hunk 1]
```diff
@@ -2627,7 +2627,7 @@ func getSecretMount(tokens []string, secrets map[string]define.Secret, mountlabe
 
 	// Copy secrets to container working dir (or tmp dir if it's an env), since we need to chmod,
 	// chown and relabel it for the container user and we don't want to mess with the original file
-	if err := os.MkdirAll(filepath.Dir(ctrFileOnHost), 0644); err != nil {
+	if err := os.MkdirAll(filepath.Dir(ctrFileOnHost), 0755); err != nil {
 		return nil, "", err
 	}
 	if err := ioutil.WriteFile(ctrFileOnHost, data, 0644); err != nil {
```
Comment (ashley-cui) [inline run_linux.go:2633]: Do we also need to change the perms on the file itself or just the dir?

Comment (rhatdan) [reply run_linux.go:2633]: File is fine, it is world readable, This issue is 644 means a direcory it listable but not searchable.

So a non root user would not be able to get to the data file on disk.  Although they would be able to list it.

**Comment by ashley-cui** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@rhatdan I think the overlay test failure might be real.  It looks like a perm issue trying to get to a directory.

**Comment by TomSweeneyRedHat** [comment]
Code LGTM, maybe a test case adjustment is needed?

**Review by flouthoc — APPROVED** [review]

**Pull request merged**